packages/gatsby-plugin-utils/src/test-plugin-options-schema.ts

```diff
@@ -12,7 +12,35 @@ export async function testPluginOptionsSchema(
   pluginSchemaFunction: Exclude<GatsbyNode["pluginOptionsSchema"], undefined>,
   pluginOptions: IPluginInfoOptions
 ): Promise<ITestPluginOptionsSchemaReturnType> {
-  const pluginSchema = pluginSchemaFunction({ Joi })
+  const pluginSchema = pluginSchemaFunction({
+    Joi: Joi.extend(joi => {
+      return {
+        base: joi.any(),
+        type: `subPlugins`,
+        args: (): any =>
+          joi
+            .array()
+            .items(
+              joi
+                .alternatives(
+                  joi.string(),
+                  joi.object({
+                    resolve: Joi.string(),
+                    options: Joi.object({}).unknown(true),
+                  })
+                )
+                .custom(value => {
+                  if (typeof value === `string`) {
+                    value = { resolve: value }
+                  }
+
+                  return value
+                }, `Gatsby specific subplugin validation`)
+            )
+            .default([]),
+      }
+    }),
+  })
```

Comment:
test helper wasn't aware of `.subPlugins` so this adds smaller stub that pretty much just checks for valid structure